openqabot/utils.py

```diff
@@ -21,6 +22,22 @@ def create_logger(name: str) -> logging.Logger:
     return log
 
 
+def get_yml_list(path: Path) -> List[Path]:
+    """
+    Create a list of YML filenames from a folder
+    or from a single file path.
+    """
+    yml_list = []
+    extensions = ("yml", "yaml")
+    for ext in extensions:
+        if path.is_file():
+            if path.name.endswith("." + ext):
+                yml_list.append(path)
+        else:
+            yml_list += list(path.glob("*." + ext))
```

Comment:
```suggestion
    extensions = (".yml", ".yaml")
    if path.is_file() and path.suffix in extensions:
        yml_list.append(path)
    elif path.is_dir()
        yml_list += [f for f in p.iterdir() if f.suffix in extensions]
```

Reply:
I like to keep the `if` in two separate layers like

```
 if path.is_file():
            if path.name.endswith("." + ext):
```

and not doing 
```
 if path.is_file() and path.name.endswith("." + ext):
```

for a specific use case. It is the one covered by unit test `test_get_yml_list_single_file_not_yml` : th euser provide a single file path (not a folder) but with an invalid extension. Using the `if  ... AND ... :` pattern results in it to be considered a folder.

Reply:
I like to change `else:` to `elif path.is_dir():` 

I'm not sure I like to move away from `.glob` that was the original code also before https://github.com/openSUSE/qem-bot/pull/133